packages/contracts/contracts/foundry/Sphinx.sol

```diff
@@ -124,14 +127,30 @@ abstract contract Sphinx {
             revert("Incorrect execution type.");
         }
 
-        DeploymentInfo memory deploymentInfo = sphinxCollect(_executionMode, deployer);
+        DeploymentInfo memory deploymentInfo = sphinxCollect(
+            _executionMode,
+            deployer,
+            _systemContractsFilePath
+        );
         vm.writeFile(_deploymentInfoPath, vm.toString(abi.encode(deploymentInfo)));
     }
 
     function sphinxCollect(
         ExecutionMode _executionMode,
-        address _executor
+        address _executor,
+        string memory _systemContractsFilePath
     ) private returns (DeploymentInfo memory) {
+        SystemContractInfo[] memory systemContracts = abi.decode(
+            vm.parseBytes(vm.readFile(_systemContractsFilePath)),
+            (SystemContractInfo[])
+        );
```

Comment:
If you're wondering why we read/write these to/from the file system, see [these docs](https://github.com/sphinx-labs/sphinx/pull/1403/files#diff-a10a154e6c3c80ec7f19d201fb8f80e05be0abc3fe778cfeca1675cfe95cbd19R1117-R1124). (This code was previously in `Sphinx.sol:sphinxEstimateMerkleLeafGas`

Reply:
From the `write-initcode.ts` docs:
```
Including it in the production Foundry plugin would *significantly* slow down the user's compilation process if they're using Yul (i.e. `viaIR`) with the optimizer enabled.
```

If I remember correctly, it took minutes to compile the user's script. It was significant enough to warrant a messier pattern like this. 

In the future, I think we should move this to the TypeScript code that's executed before we collect the user's transactions